tests/main/views/test_direct_award.py

```diff
@@ -548,6 +548,103 @@ def test_award_contract_raises_400_if_project_is_not_locked(self):
         assert res.status_code == 400
 
 
+class TestDirectAwardTellUsAboutContract(TestDirectAwardBase):
+    url = '/buyers/direct-award/g-cloud/projects/1/tell-us-about-contract'
+
+    @pytest.fixture
+    def client(self):
+        self.login_as_buyer()
+        return self.client
+
+    @pytest.fixture
+    def xpath(self, client):
+        res = client.get(self.url)
+        doc = html.fromstring(res.get_data(as_text=True))
+        return doc.xpath
+
+    @pytest.fixture
+    def data(self):
+        return {
+            'start_date-day': '31',
+            'start_date-month': '12',
+            'start_date-year': '2019',
+            'end_date-day': '1',
+            'end_date-month': '6',
+            'end_date-year': '2021',
+            'buying_organisation': 'Lewisham Council',
+            'value_in_pounds': '100.00'
+        }
+
+    def test_tell_us_about_contract_exists(self, client):
+        assert client.get(self.url).status_code == 200
+
+    def test_tell_us_about_contract_renders(self, xpath):
+        assert xpath('boolean(//h1[contains(normalize-space(), "Tell us about your contract")])')
+
+    def test_tell_us_about_contract_form_fields(self, xpath):
+        assert xpath('count(//input[@type="text"][substring-after(@name, "start_date")])') == 3
+        assert xpath('count(//input[@type="text"][substring-after(@name, "end_date")])') == 3
+        assert xpath('boolean(//input[@type="text"][contains(@name, "value_in_pounds")])')
+        assert xpath('boolean(//input[@type="text"][contains(@name, "buying_organisation")])')
+        assert xpath('boolean(//input[@type="submit"][@value="Submit"])')
+
+    def test_previous_page_button_exists_and_points_to_previous_page(self, xpath):
+        assert xpath(
+            'boolean('
+            '//a[@href="/buyers/direct-award/g-cloud/projects/1/which-service-won-contract"]'
+            '[contains(normalize-space(), "Previous page")])')
+
+    def test_tell_us_about_contract_form_action_url_is_tell_us_about_contract_url(self, xpath):
+        assert xpath('//form[@class="tell-us-about-contract-form"]/@action')[0] == self.url
+
+    def test_tell_us_about_contract_raises_400_if_project_is_not_accessible(self, client):
+        self.data_api_client.get_direct_award_project.side_effect = HTTPError(mock.Mock(status_code=400))
```

Comment:
Feels like there might be something wrong here - the other views use `is_direct_award_project_accessible`, and throw a 404 if it isn't. This view doesn't do that... but somehow the test passes. Hmm!

Reply:
This needs fixing